src/interfaces.ts

```diff
@@ -85,3 +85,123 @@ export interface ITileIntensity {
   y: number
   probability: number
 }
+
+/**
+ * Interface used for goals and absorptions
+ */
+export interface IAbsorption {
+  x: number
+  y: number
+  probability: number
+}
```

Comment:
`Photons` interfaces should be in Photons, not here. Otherwise, it is a mess of low-level and high-level operations.